ms2query/ms2library.py

```diff
@@ -0,0 +1,332 @@
+from ms2query.app_helpers import load_pickled_file
+from ms2query.spectrum_processing import create_spectrum_documents
+
+
+class Ms2Library:
+    """Calculates scores of spectra in library and selects best matches"""
+    def __init__(self,
+                 sqlite_file_location: str,
+                 s2v_model_file_name: str,
+                 ms2ds_model_file_name: str,
+                 pickled_s2v_embeddings_file_name: str,
+                 pickled_ms2ds_embeddings_file_name: str,
+                 **settings):
+        """
+        Parameters
+        ----------
+        sqlite_file_location:
+            The location at which the sqlite_file_is_stored. The file is
+            expected to have 3 tables: tanimoto_scores, inchikeys and
+            spectra_data.
+        s2v_model_file_name:
+            File location of a spec2vec model. In addition two more files in
+            the same folder are expected with the same name but with extensions
+            .trainables.syn1neg.npy and .wv.vectors.npy.
+        ms2ds_model_file_name:
+            File location of a trained ms2ds model.
+        pickled_s2v_embeddings_file_name:
+            File location of a pickled file with Spec2Vec embeddings in a
+            pd.Dataframe with as index the spectrum id.
+        pickled_ms2ds_embeddings_file_name:
+            File location of a pickled file with ms2ds embeddings in a
+            pd.Dataframe with as index the spectrum id.
+
+        **settings:
+            As additional parameters predefined settings can be changed.
+        spectrum_id_column_name:
+            The name of the column or key in dictionaries under which the
+            spectrum id is stored. Default = "spectrumid"
+        cosine_score_tolerance:
+            Setting for calculating the cosine score. If two peaks fall within
+            the cosine_score tolerance the peaks are considered a match.
+            Default = 0.1
+        base_nr_mass_similarity:
+            The base nr used for normalizing the mass similarity. Default = 0.8
+        max_parent_mass:
+            The value used to normalize the parent mass by dividing it by the
+            max_parent_mass. Default = 13428.370894192036
+        """
+        # pylint: disable=too-many-arguments
+        # Change default settings to values given in **settings
+        settings = self._set_settings(settings)
+
+        # Set given settings
+        self.spectrum_id_column_name = settings["spectrum_id_column_name"]
+        # todo create a ms2query model class that stores the model but also the
+        #  settings used, since the settings used should always be the same as
+        #  when the model was trained
+        self.cosine_score_tolerance = settings["cosine_score_tolerance"]
+        self.base_nr_mass_similarity = settings["base_nr_mass_similarity"]
+        # todo make new model that has a fixed basic mass
+        self.max_parent_mass = settings["max_parent_mass"]
```

Comment:
This part could be removed entirely when we switch to `self.settings = self._set_settings(settings)`. 
Instead of later calling `self.spectrum_id_column_name`, it would then be `self.settings["spectrum_id_column_name"]`

Reply:
Good suggestion! Makes the init function a lot clearer, did the same for create_library_files.py